### PR TITLE
plugins/friendly-snippets: switch to mkVimPlugin

### PR DESCRIPTION
**Pull request by GaetanLepage**

### CODE REVIEW

File: plugins/by-name/friendly-snippets/default.nix [hunk 1]
```diff
@@ -1,31 +1,13 @@
...
-    enable = mkEnableOption "friendly-snippets";
-
-    package = lib.mkPackageOption pkgs "friendly-snippets" {
-      default = [
-        "vimPlugins"
-        "friendly-snippets"
-      ];
-    };
-  };
+lib.nixvim.vim-plugin.mkVimPlugin {
```
Comment (khaneliman) [inline plugins/by-name/friendly-snippets/default.nix:5]: This is a lua plugin, though https://github.com/rafamadriz/friendly-snippets

Comment (GaetanLepage) [reply plugins/by-name/friendly-snippets/default.nix:5]: Yes, but as it is not configured using lua code, we don't consider it as a "neovim" plugin.
Plugins that have no configuration are implemented using `mkVimPlugin`.

Maybe long term we can find a better abstraction. In the short-term we'll stick with this methodology.

Comment (khaneliman) [reply plugins/by-name/friendly-snippets/default.nix:5]: We've been using `hasSettings = false` and `callSetup = false`

Comment (GaetanLepage) [reply plugins/by-name/friendly-snippets/default.nix:5]: Sure, but this makes not much more sense to me...
The `mkNeovimPlugin` name is a bit misleading, it should be `mkPluginThatIsConfiguredWithRequireLuaOrSomethingSimilar`.

I am open to discussing how to rethink those helpers and their respective field.
However, using `mkVimPlugin` here is fine in my opinion.

**Comment by khaneliman** [comment]
Want to cleanup the entry in https://github.com/nix-community/nixvim/pull/2681

**Comment by GaetanLepage** [comment]
> Want to cleanup the entry in #2681

Good point, done

File: plugins/by-name/friendly-snippets/default.nix (hunk 1 above)
Comment (MattSturgeon) [reply plugins/by-name/friendly-snippets/default.nix:5]: > We've been using `hasLuaConfig = false`, `hasSettings = false`, and `callSetup = false`

The fact we can now use either `mkNeovimPlugin` or `mkVimPlugin` makes me wonder about combining them together into on single function?

Or alternatively, maybe a lot of the modes available via args should actually be special-case wrapper functions that wrap a single lower-level `mkPluginWith`?

**Review by MattSturgeon — APPROVED** [review]

**Comment by GaetanLepage** [comment]
@mergifyio queue

**Pull request merged**